compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertLinalgMatmulToMmt4D.cpp

```diff
@@ -331,45 +332,58 @@ LinalgMatmulOpToLinalgMmt4DOpPattern::chooseTileParams(Value lhs, Value rhs,
   int64_t shapeN = rhsType.getShape()[1];
   auto chooseMatMulOrMatVec = [=](ArrayRef<int> m0k0n0,
                                   ArrayRef<int> m0k0n0ForMatVec,
+                                  ArrayRef<int> m0k0n0ForWhenRhsHas2Columns,
                                   std::string comment) {
     assert(m0k0n0ForMatVec[2] == 1 && "not a matrix*vector shape");
-    if (shapeN == 1) {
-      return Mmt4DTileParams(m0k0n0ForMatVec, comment + ", matrix*vector");
-    } else if (shapeM == 1) {
+    assert(m0k0n0ForWhenRhsHas2Columns[2] == 2 && "not a matrix*vector2 shape");
+
+    SmallVector<int> config;
+    if (shapeN == 1 || shapeM == 1) {
+      config.assign(m0k0n0ForMatVec.begin(), m0k0n0ForMatVec.end());
+    } else if (shapeN == 2 || shapeM == 2) {
+      config.assign(m0k0n0ForWhenRhsHas2Columns.begin(),
+                    m0k0n0ForWhenRhsHas2Columns.end());
+    } else {
+      return Mmt4DTileParams(m0k0n0, comment);
+    }
+
+    if (shapeN == 1 || shapeN == 2) {
+      comment += ", matrix*vector, where the vector has " +
+                 std::to_string(shapeN) + " column(s)";
+    } else {
       // The vector*matrix case is intentionally derived from the matrix*vector
       // case by swapping M and N dims so that in kernel codegen we can reuse
       // matrix*vector kernels by swapping LHS and RHS.
-      int m0k0n0ForVecMat[3] = {m0k0n0ForMatVec[2], m0k0n0ForMatVec[1],
-                                m0k0n0ForMatVec[0]};
-      return Mmt4DTileParams(m0k0n0ForVecMat, comment + ", vector*matrix");
-    } else {
-      return Mmt4DTileParams(m0k0n0, comment);
+      std::reverse(config.begin(), config.end());
```

Comment:
Maybe `std::swap` on the individual M, N entries is more lightweight and explicit?